fiftyone/core/view.py

```diff
@@ -1637,66 +1637,39 @@ def _pipeline(
         if _attach_frames_idx is None and (attach_frames or frames_only):
             _attach_frames_idx = len(_pipelines)
 
-        #######################################################################
-        # Insert frame lookup pipeline(s) if needed
-        #######################################################################
-
         if _attach_frames_idx1 is not None and _attach_frames_idx is not None:
             _attach_frames_idx = _attach_frames_idx1
 
-        if _attach_frames_idx0 is not None and _attach_frames_idx is not None:
-            # Two lookups are required; manually do the **last** one and rely
-            # on dataset._pipeline() to do the first one
-            attach_frames = True
-            _pipeline = self._dataset._attach_frames_pipeline(
-                limit=limit_frames, support=support
-            )
-            _pipelines.insert(_attach_frames_idx, _pipeline)
-        elif _found_flattened_videos and _attach_frames_idx is not None:
-            # Must manually attach frames after the group $lookup
-            attach_frames = None  # special syntax: frames already attached
-            _pipeline = self._dataset._attach_frames_pipeline(
-                limit=limit_frames, support=support
-            )
-            _pipelines.insert(_attach_frames_idx, _pipeline)
-        elif _attach_frames_idx0 is not None or _attach_frames_idx is not None:
-            # Exactly one lookup is required; rely on dataset._pipeline() to
-            # do it
-            attach_frames = True
+        _inserts = []
 
-        # @todo use the optimization below instead, which injects frames as
```

Comment:
@benjaminpkane I first tried to make this optimization awhile ago, but, as documented here, when I initially tried it, there was some App rendering issue where poster frames would not show labels.

However, I am not seeing any issues with first-frame label rendering now, so it seems like whatever the previous limitation was has now been lifted! 🤞🤞🤞

A couple questions for you:
- Do you know what might have changed to resolve the symptom reported in this `@todo`?
- Do you foresee any issues with `$lookup` potentially occurring later in pipelines now?

Reply:
It is likely due to https://github.com/voxel51/fiftyone/pull/5450. I don't see any issue with this PR. Though it looks like some tests need to be updated

Reply:
Good catch, test has been updated!